OutOfSchool/OutOfSchool.WebApi/Services/Images/ImageDependentEntityImagesInteractionMediator.cs

```diff
@@ -21,152 +21,168 @@ namespace OutOfSchool.WebApi.Services.Images
     // TODO: make synchronization to remove incorrect operations' ids
 
     /// <summary>
-    /// Represents a base class for operations with images.
+    /// Represents a mediator class for operations with images.
+    /// This instance does not save the given entity changes.
     /// </summary>
     /// <typeparam name="TEntity">Entity type.</typeparam>
-    /// <typeparam name="TKey">The type of entity Id.</typeparam>
-    public class ImageInteractionBaseService<TEntity, TKey> : IImageInteractionService<TKey>
-        where TEntity : class, IKeyedEntity<TKey>, IImageDependentEntity<TEntity>, new()
+    internal class ImageDependentEntityImagesInteractionMediator<TEntity> : IImageDependentEntityImagesInteractionMediator<TEntity>
```

Comment:
Not a call to action, just consideration.
It's definitely not a [Mediator pattern](https://refactoring.guru/design-patterns/mediator). 
Could other name be chosen (and why `service` is not good)?
Could all logic (ImageService + this Mediator) be in one `service` that just implements 2 different interfaces?